zcash_client_sqlite/src/wallet.rs

```diff
@@ -2176,15 +2184,11 @@ mod tests {
 
         // Expire the shielding transaction.
         let expiry_height = st.wallet().get_transaction(txid).unwrap().expiry_height();
-        st.wallet_mut().update_chain_tip(expiry_height).unwrap();
+        st.wallet_mut().update_chain_tip(expiry_height + 1).unwrap();
```

Comment:
```suggestion
        st.wallet_mut().update_chain_tip(expiry_height).unwrap();
```
This was correct. If we update the chain tip height to the expiry height then the tx is expired, because that was the last block height it could be mined at.

Reply:
If we make this change first it should fail, and then it will be fixed by correcting the off-by-ones in the SQL queries.